utils/python/tests/scripts_regression_tests.py

```diff
@@ -1705,6 +1705,34 @@ def xml_to_tester(self, xml_string):
         test_xml = _wrap_config_build_xml(xml_string)
         return CMakeTester(self, get_macros(self._maker, test_xml, "CMake"))
 
+###############################################################################
+class S_TestManageAndQuery(unittest.TestCase):
+    """Tests various scripts to manage and query xml files"""
+
+    def _run_and_assert_query_testlist(self, extra_args=""):
+        """Ensure that query_testlist runs successfully with the given extra arguments"""
+
+        testlist_allactive = os.path.join(CIME.utils.get_model_config_root(), "allactive", "testlist_allactive.xml")
+
+        run_cmd_assert_result(self, "%s/query_testlists --xml-testlist %s %s"%
+                              (SCRIPT_DIR, testlist_allactive, extra_args))
+
+    def test_query_testlists_runs(self):
+        """Make sure that query_testlists runs successfully
+
+        This simply makes sure that query_testlists doesn't generate any errors
+        when it runs. This helps ensure that changes in other utilities don't
+        break query_testlists.
+        """
+        self._run_and_assert_query_testlist()
```

Comment:
I generally like to do more than simple "smoke" tests like this, but these simple smoke tests seemed better than nothing, and I was having trouble justifying spending the time writing more detailed tests for this less-critical code.